pkg/job/factory.go

```diff
@@ -99,15 +100,23 @@ func ConstructDockerJob( //nolint:funlen
 		return &model.Job{}, err
 	}
 	j.APIVersion = a.String()
-
+	//If entrypoint is specified, construct a 1 element slice for it. Otherwise don't pass anything for that field
+	var entrypointSlice []string
+	if entrypoint == "" {
+		//tell docker to use default entrypoint, which can be defined in the Dockerfile
+		entrypointSlice = nil
+	} else {
+		entrypointSlice = []string{entrypoint}
+	}
```

Comment:
`entrypointSlice` will be nil by default, so you only need to handle the `else` case, e.g. `if entrypoint != ""`